lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1346,6 +1346,21 @@ void DifferentiableActivityInfo::analyze(DominanceInfo *di,
                 setVaried(result, i);
           }
         }
+        // Handle `struct_element_addr`.
+        else if (auto *seai = dyn_cast<StructElementAddrInst>(&inst)) {
```

Comment:
The body is exactly the same as SEI's except for `seai`/`sei`. Could you write a macro?

Reply:
Macro'd in https://github.com/apple/swift/pull/22973/commits/272e986ca9dbf0faaa6040e24b0d34d57c9188c5: `#define DIAGNOSE_STRUCT_NO_DERIVATIVE_FIELD(INST)`.